cmd/init.go

```diff
@@ -122,3 +139,24 @@ ks init app-name --api-spec=version:v1.7.1
 # specific build of Kubernetes to generate 'ksonnet-lib'.
 ks init app-name --api-spec=file:swagger.json`,
 }
+
+func genKsRoot(appName, appDir, ksDir string) (string, error) {
+	if appDir == "" {
+		return "", errors.New("invalid application directory")
+	}
+
+	if appName == "" && ksDir == "" {
+		return "", errors.New("invalid application name")
+	}
+
+	if ksDir != "" {
+		if filepath.IsAbs(ksDir) {
+			return ksDir, nil
+		}
+
+		path := fmt.Sprintf("%s%s%s", appDir, string(filepath.Separator), ksDir)
```

Comment:
Probably better to use `return filepath.Join(appDir, ksDir)` here in order to not deal with trailing slashes